lib/auth/auth_with_roles.go

```diff
@@ -3453,6 +3495,22 @@ func (a *ServerWithRoles) UpdateAndSwapUser(ctx context.Context, user string, wi
 	return nil, trace.NotImplemented("func UpdateAndSwapUser is not implemented by ServerWithRoles")
 }
 
+func (a *ServerWithRoles) checkOktaUpdateAccess(ctx context.Context, new types.User, existing string) error {
+	if a.hasBuiltinRole(types.RoleOkta) {
```

Comment:
I considered the conversation here further: https://github.com/gravitational/teleport/pull/33630#discussion_r1382028202

I still think we need to address the concerns here.  Specifically:
1) As discussed above I think there is concerned about the user access state getting out of sync which could result in Okta clobbering other changes
2) We must validate that the Okta origin is not being added in update.  Otherwise this is a potential bypass to allow non-Okta users to now be modified by the Okta user (with collaboration for the initial update)